homeassistant/auth/providers/trusted_networks.py

```diff
@@ -136,13 +159,22 @@ async def async_get_or_create_credentials(
 
         users = await self.store.async_get_users()
         for user in users:
-            if not user.system_generated and user.is_active and user.id == user_id:
-                for credential in await self.async_credentials():
-                    if credential.data["user_id"] == user_id:
-                        return credential
-                cred = self.async_create_credentials({"user_id": user_id})
-                await self.store.async_link_user(user, cred)
-                return cred
+            if user.id != user_id:
```

Comment:
I rewrote the if-for-if spaghetti to individual statements to make it easier to read.